strings/longest_substr_no_repeating.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+* @file
+* @brief Sliding window solution to find the longest substring with no repeating characters
+* @details
+    * Given a string we are required to find the
+    * longest possible substring without any repeating characters.
+    * The time and space complexity are both O(n) since an unordered set is used
+    * and the entire string is traversed.
+* @author [Saketh Renangi](https://github.com/rensaketh)
+*/
+
+#include <cassert>   /// for assert
+#include <unordered_set>  /// for unordered_set
+#include <string> //for string
+
+/**
+ * @brief This function uses the sliding window approach to
+ * find the longest substring that does not have a repeating character
+ *
+ * @param String named s, used to find substrings from this string
+ *
+ * @returns Retuns an integer that is the length of the longest substring
+ * with no repeated characters
+ */
+
+int lengthOfLongestSubstring(std::string s) {
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂